### PR TITLE
Added support for Windows 2022 nodes

### PR DESCRIPTION
**Pull request by xrow**
##### SUMMARY

The feature adds the capability to create Windows 2022 nodes via the os_sku property.

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
azure_rm_aks

### CODE REVIEW

File: plugins/modules/azure_rm_aksagentpool.py [hunk 1]
```diff
@@ -361,6 +375,10 @@ def __init__(self):
                 type='str',
                 choices=['Linux', 'Windows']
             ),
+            os_sku=dict(
```
Comment (Fred-sun) [inline plugins/modules/azure_rm_aksagentpool.py:825]: The SDK only supports two choices for this parameters ("Ubuntu", "CBLMariner")

File: plugins/modules/azure_rm_aks.py [hunk 2]
```diff
@@ -676,6 +720,7 @@ def create_addon_profiles_spec():
     vnet_subnet_id=dict(type='str'),
     availability_zones=dict(type='list', elements='int', choices=[1, 2, 3]),
     os_type=dict(type='str', choices=['Linux', 'Windows']),
+    os_sku=dict(type='str', choices=['Ubuntu', 'Windows2019', 'Windows2022']),
```
Comment (Fred-sun) [inline plugins/modules/azure_rm_aks.py]: The SDK only supports two choices for this parameters ("Ubuntu", "CBLMariner")

File: plugins/modules/azure_rm_aks.py [hunk 3]
```diff
@@ -152,6 +167,14 @@
...
                     - Linux
                     - Windows
+            os_sku:
+                description:
+                    - The operating system sku.
+                type: str
+                choices:
+                    - Ubuntu
+                    - Windows2019
+                    - Windows2022
```
Comment (Fred-sun) [inline plugins/modules/azure_rm_aks.py]: trailing whitespace
```suggestion
                    - Windows2022
```

File: plugins/modules/azure_rm_aksagentpool.py [hunk 4]
```diff
@@ -64,6 +64,14 @@
...
             - Linux
             - Windows
+    os_sku:
+        description:
+            - The operating system sku.
+        type: str
+        choices:
+            - Ubuntu
+            - Windows2019
+            - Windows2022
```
Comment (Fred-sun) [inline plugins/modules/azure_rm_aksagentpool.py]:
```suggestion
            - Windows2022
```

File: plugins/modules/azure_rm_aks.py [hunk 5]
```diff
@@ -657,6 +697,10 @@ def create_addon_profiles_spec():
     ssh_key=dict(type='str', no_log=True, required=True)
 )
 
+windows_profile_spec = dict(
+    admin_username=dict(type='str', required=True),
+    admin_password=dict(type='str', required=True),
```
Comment (Fred-sun) [inline plugins/modules/azure_rm_aks.py]: admin_password could be a secret
```suggestion
    admin_password=dict(type='str', no_log=True, required=True),
```

**Review by Fred-sun — COMMENTED** [review]
small change request！

File: plugins/modules/azure_rm_aks.py (hunk 2 above)
Comment (xrow) [reply plugins/modules/azure_rm_aks.py]: @Fred-sun 

Are you sure that only 2 params are supported? It did work for me... to create windows nodes.

https://github.com/Azure/azure-sdk-for-python/blob/81fa2f833163808415256a0f6722d0a023cf50eb/sdk/containerservice/azure-mgmt-containerservice/azure/mgmt/containerservice/v2023_04_01/models/_container_service_client_enums.py#L294

https://github.com/Azure/azure-sdk-for-python/blob/81fa2f833163808415256a0f6722d0a023cf50eb/sdk/containerservice/azure-mgmt-containerservice/azure/mgmt/containerservice/v2021_07_01/models/_models_py3.py#L2509

It looks like any string is valid too! Right?

Do you think it is correct to alter the list to all types from https://github.com/Azure/azure-sdk-for-python/blob/81fa2f833163808415256a0f6722d0a023cf50eb/sdk/containerservice/azure-mgmt-containerservice/azure/mgmt/containerservice/v2023_04_01/models/_container_service_client_enums.py#L294

**Comment by Fred-sun** [comment]
@xrow What is your version of azure-mgmt-containerservice?

**Comment by xrow** [comment]
@Fred-sun I believe it was using latest, but i acutally do not know. I do not know how to test.

**Comment by Fred-sun** [comment]
@xrow I am currently using the version of "azure-mgmt-containerservice==20.0.0", I will upgrade to the latest version test, thank you!

**Comment by xrow** [comment]
@Fred-sun Based on your feedback I added AzureLinux as a sku too. I tought that is the most reasonable todo.

File: plugins/modules/azure_rm_subnet.py [hunk 6]
```diff
@@ -343,16 +343,6 @@
     serviceName=dict(
         type='str',
         required=True,
-        choices=['Microsoft.Web/serverFarms', 'Microsoft.ContainerInstance/containerGroups', 'Microsoft.Netapp/volumes',
```
Comment (Fred-sun) [inline plugins/modules/azure_rm_subnet.py]: Why delete the choices？  Please keep！

Comment (xrow) [reply plugins/modules/azure_rm_subnet.py]: @Fred-sun I am sorry. That is actually part of a different pull request I am planning. Let me figure out on how to have multipole pulls at the same time. Can you mean comment of the remaining open topics? I will revert this part later. Sorry again, I am no so strong with pull requests and its management.

Comment (xrow) [reply plugins/modules/azure_rm_subnet.py]: @Fred-sun Hi, I moved the change into a different pull https://github.com/ansible-collections/azure/pull/1414  .. We can go ahead on this one.

**Comment by Fred-sun** [comment]
@xrow Other os_sku options will only be supported in versions higher than azure-mgmt-containerservice=20.0.0. I will upgrade azure-mgmt-containerservice first, and then push forward the PR merge in testing. Thank you!

**Comment by xrow** [comment]
Thank you too...

**Comment by Fred-sun** [comment]
@xrow This branch has conflicts, Please help to resolved. The relate package has been upgrade. when you resolved, I will review and push for merged as soon as possible! Thank you very much!

**Comment by xrow** [comment]
@Fred-sun  can you review again?

File: plugins/modules/azure_rm_aksagentpool.py [hunk 7]
```diff
@@ -615,6 +623,12 @@
             type: str
             returned: always
             sample: Linux
+        os_sku:
```
Comment (Fred-sun) [inline plugins/modules/azure_rm_aksagentpool.py:626]: duplication of key "os_sku" in mapping ( line 717)

File: plugins/modules/azure_rm_aksagentpool.py [hunk 8]
```diff
@@ -64,6 +64,14 @@
         choices:
             - Linux
             - Windows
+    os_sku:
```
Comment (Fred-sun) [inline plugins/modules/azure_rm_aksagentpool.py:67]: duplication of key "os_sku" in mapping( line 127)

File: plugins/modules/azure_rm_aksagentpool.py [hunk 9]
```diff
@@ -808,6 +822,10 @@ def __init__(self):
                 type='str',
                 choices=['Linux', 'Windows']
             ),
+            os_sku=dict(
```
Comment (Fred-sun) [inline plugins/modules/azure_rm_aksagentpool.py:825]: duplication of key "os_sku" in mapping(line 861)

**Comment by Fred-sun** [comment]
kindly ping!

**Comment by xrow** [comment]
@Fred-sun Hi, Fred I am sorry I am not allowed to dedicate time on this any more by the organisation I am working for. I guess you have to take it from here or close it. I am sorry for it.

**Comment by Fred-sun** [comment]
@xrow Thank you for your feedback, you can authorize me to maintain this PR together, thank you!

**Comment by xrow** [comment]
@Fred-sun Please do so! Thank you.

**Comment by Fred-sun** [comment]
@xrow Support in #1740

**Comment by xrow** [comment]
ty

**Pull request closed without merging**